event-driven-architecture/src/main/java/com/iluwatar/eda/framework/DynamicRouter.java

```diff
@@ -0,0 +1,13 @@
+package com.iluwatar.eda.framework;
+
+/**
+ * A {@link DynamicRouter} is responsible for selecting the proper path of a {@link Message}
+ * Messages can be associated to Channels through the registerChannel method and dispatched by
+ * calling the dispatch method.
+ */
+public interface DynamicRouter<E extends Message> {
```

Comment:
I have removed `DynamicRouter` interface. After some thought don't think it's relevant that much to demonstrate the pattern.